overlord/servicestate/quota_handlers.go

```diff
@@ -322,22 +325,26 @@ type ensureSnapServicesForGroupOptions struct {
 // the same changes to be processed and nothing will be broken. This is mainly
 // a consequence of calling wrappers.EnsureSnapServices().
 // Currently, it only supports handling a single group change.
-func ensureSnapServicesForGroup(st *state.State, t *state.Task, grp *quota.Group, opts *ensureSnapServicesForGroupOptions, meter progress.Meter, perfTimings *timings.Timings) error {
+func ensureSnapServicesForGroup(st *state.State, t *state.Task, grp *quota.Group, opts *ensureSnapServicesForGroupOptions, perfTimings *timings.Timings) error {
 	if opts == nil {
 		return fmt.Errorf("internal error: unset group information for ensuring")
 	}
 
 	allGrps := opts.allGrps
 
-	if meter == nil {
-		meter = progress.Null
+	var meterLocked, meterUnlocked progress.Meter
+	if t == nil {
+		meterLocked = progress.Null
+		meterUnlocked = progress.Null
+	} else {
+		meterUnlocked = snapstate.NewTaskProgressAdapterUnlocked(t)
+		meterLocked = snapstate.NewTaskProgressAdapterUnlocked(t)
```

Comment:
this should be s/Unlocked/Locked/. It's probably a bit annoying to check for this in tests:/

Reply:
this error actually comes from https://github.com/snapcore/snapd/pull/10410, but you're right it's wrong... I will adjust that there. like you said though, not sure how to really test the different meters